src/Vocup/IO/Vhf2/Vhf2Format.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Vocup.Models;
+
+namespace Vocup.IO.Vhf2;
+
+public class Vhf2Format : BookFileFormat
+{
+    private const string headerFileName = "VOCUP VOCABULARY BOOK";
+    private const string bookFileName = "book.2.json";
+    private readonly JsonSerializerOptions options;
+    private readonly Version fileVersion;
+
+    public static Vhf2Format Instance { get; } = new();
+
+    private Vhf2Format()
+    {
+        options = new JsonSerializerOptions
+        {
+            PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+            WriteIndented = true
+        };
+        options.Converters.Add(new PracticeModeConverter());
+        options.Converters.Add(new PracticeResultConverter());
+
+        fileVersion = new Version(2, 0);
+    }
+
+    public void Read(FileStream stream, VocabularyBook book)
+    {
+        try
+        {
+            using ZipArchive archive = new(stream, ZipArchiveMode.Read, leaveOpen: true);
+
+            ZipArchiveEntry versionFile = archive.GetEntry(headerFileName)
```

Comment:
Making the header a JSON file, too is a good idea. I did not want to include the version in the book JSON because then I would need to partially deserialize it. First deserialize only the version, compare it, reset the stream and then deserialize the entire file.